File: Assignments/Assignment1/TranDinhDuyPhuong_SE18D04/src/services/categoryService.js
import api from './api';

const getAllCategories = async () => {
    const response = await api.get('/categories');
    return response.data?.data || [];
};

const getCategoryById = async (id) => {
    const response = await api.get(`/categories/${id}`);
    return response.data;
};

const createCategory = async (categoryData) => {
    const response = await api.post('/categories', categoryData);
    return response.data;
};

const updateCategory = async (id, categoryData) => {
    const response = await api.put(`/categories/${id}`, categoryData);
    return response.data;
};

const deleteCategory = async (id) => {
    const response = await api.delete(`/categories/${id}`);
    return response.data;
};

const searchCategories = async (keyword) => {
    const response = await api.get(`/categories/search?keyword=${keyword}`);
    return response.data;
};

const categoryService = {
    getAllCategories,
    getCategoryById,
    createCategory,
    updateCategory,
    deleteCategory,
    searchCategories
};

export default categoryService;

File: Assignments/Assignment2/A2TranDinhDuyPhuong_SE18D04/src/main/java/sba301/a2trandinhduyphuong_se18d04/services/CategoryService.java
package sba301.a2trandinhduyphuong_se18d04.services;

import sba301.a2trandinhduyphuong_se18d04.dtos.request.CategoryRequest;
import sba301.a2trandinhduyphuong_se18d04.dtos.response.CategoryResponse;

import java.util.List;

public interface CategoryService {

    List<CategoryResponse> getAllCategories();

    CategoryResponse getCategoryById(Long id);

    CategoryResponse createCategory(CategoryRequest request);

    CategoryResponse updateCategory(Long id, CategoryRequest request);

    void deleteCategory(Long id);

    List<CategoryResponse> searchCategories(String keyword);

}
